### PR TITLE
rubocop: use new cops and take autocorrect for violations

### PR DESCRIPTION
**Pull request by ndushay**
@jcoyne it seems wrong to have Gemfile.lock checked into git -- was that a mistake?  (Also, why use Gemfile instead of development dependencies in the gemspec?)

## Why was this change made?

 reduce noise for next sprint

## How was this change tested?

circleci

### CODE REVIEW

**Comment by jcoyne** [comment]
> it seems wrong to have Gemfile.lock checked into git 

Rails is a gem and it has a Gemfile.lock (see https://github.com/rails/rails/pull/18992)

**Comment by ndushay** [comment]
@jcoyne works for me.  thanks.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 AllCops:
-  TargetRubyVersion: 2.7
+  TargetRubyVersion: 2.7 # dor-service-app is < 3.0 due to fedora3
```
Comment (jcoyne) [inline .rubocop.yml:2]: 👍

File: datacite.gemspec [hunk 2]
```diff
@@ -33,4 +33,5 @@ Gem::Specification.new do |spec|
   spec.add_dependency "faraday_middleware", "~> 1.0"
   spec.add_dependency "json_schema", "~> 0.21.0"
   spec.add_dependency "zeitwerk", "~> 2.4"
+  spec.metadata["rubygems_mfa_required"] = "true"
```
Comment (jcoyne) [inline datacite.gemspec:36]: Should we discuss this at a dev planning?

Comment (ndushay) [reply datacite.gemspec:36]: sure - add it to the agenda.  I dont even know what it means;  just taking autocorrect.

Comment (jcoyne) [reply datacite.gemspec:36]: It means that you have to use multifactor authentication with rubygems to publish this gem.

Comment (ndushay) [reply datacite.gemspec:36]: I've asked this question twice in #dlss-infrastructure slack channel and no objections so far.  I think we should all be 2-auth for rubygems at this point.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**